### PR TITLE
fixed the failing test and broke out comment removal functionality. 

### PR DESCRIPTION
**Pull request by bmewing**
Also added more roxygen. This should address #77 and #78 

#78 was due to the number of spaces in the requirements file changing.

### CODE REVIEW

File: R/read_process_req_file.R [hunk 1]
```diff
@@ -54,7 +54,17 @@ identify_duplicate_reqs = function(content_df){
   }
 }
 
-remove_comments_from_req = function(content_df) {
+strip_comments = function(content){
+  #' @param content vector of characters
+  #' @return vector of chracters without comments or -r and trimmed whitespace
+  #' @details This was originally part of another function but was split out to be used
+  #' in a few different places.
+  content = gsub("\\-r.*|#.*", "", content)
```
Comment (AdamSpannbauer) [inline R/read_process_req_file.R]: A potential edge case where removing `-r` could have an adverse effect on package detection.  Admittedly, it's very edge, but a possible solution could be to have a flag parameter for removing `-r` in addition to comments?

```r
> strip_comments("20-random::randomNumbers(n = 1, col = 1)")
[1] "20"
```

**Review by AdamSpannbauer — COMMENTED** [review]
Think it all looks good, and again, big fan of the additional roxygen.

The only bit that I think needs to be discussed before merging is the `-r` removal I made above.

**Comment by bmewing** [comment]
I modified the function to only remove `-r` if it's the first non-whitespace character on the line. This handles legal R package names including -r without needing extra flags or even separate functions.

**Comment by AdamSpannbauer** [comment]
Again, this is getting pretty far into edge case world, but the updated implementation would fail if using a communitive form of the breaking case I provided before.

```r
> strip_comments("-random::randomNumbers(n = 1, col = 1) + 20")
[1] ""
```

**Comment by bmewing** [comment]
There's now a flag, `remove_additional_file` which is `TRUE` by default but if set to `FALSE` will ignore `-r`

**Comment by AdamSpannbauer** [comment]
will merge as soon as Travis gives us the 👍

**Review by AdamSpannbauer — APPROVED** [review]

**Pull request merged**